### ISSUE TITLE
In tf.Estimator training metric is not divided by the batch size

### ISSUE THREAD

**Issue by robertah** [tensorflow/tensorflow#32592]
**System information**
- TensorFlow version: TF 1.12
- Python version: 3.6

**Describe the current behavior**

<img width="345" alt="metric" src="https://user-images.githubusercontent.com/23496841/65050585-e9ddd900-d967-11e9-82b5-b8d0a04164db.png">

I am training a model with tf.Estimator and it seems that the training metric (root mean squared error) is not divided by the batch size (for both training and validation the batch size is 100). Indeed, for comparison this is the mean squared error loss for training and validation:

<img width="364" alt="loss" src="https://user-images.githubusercontent.com/23496841/65050797-450fcb80-d968-11e9-9a67-565efced4684.png">

**Code to reproduce the issue**
I am using the standard structure for `model_fn` in `tf.Estimator`. Here's the main part of the code:

``` Python
def model_fn(features, labels, mode, params):

    # model layers 
    # ...

    rmse = tf.metrics.root_mean_squared_error(labels, predictions)

    if mode == tf.estimator.ModeKeys.PREDICT:
        predictions = {
            'new_states': predictions
        }
        return tf.estimator.EstimatorSpec(mode, predictions=predictions)

    loss = tf.losses.mean_squared_error(labels, predictions)

    metrics = {'rmse': rmse}

    if mode == tf.estimator.ModeKeys.EVAL:
        return tf.estimator.EstimatorSpec(mode, loss=loss, eval_metric_ops=metrics)

    optimizer = tf.train.AdamOptimizer(learning_rate=params['learning_rate'])
   
    train_op = optimizer.minimize(loss, global_step=tf.train.get_global_step())

    return tf.estimator.EstimatorSpec(mode, loss=loss, train_op=train_op)

```

**Comment by oanush** [comment]
@robertah ,
Can you share a standalone code to reproduce the issue?thanks!

**Comment by robertah** [comment]
I can't share the original input data, but the error can be reproduced in the following way:
``` Python
import tensorflow as tf
import sys

def model_fn(features, labels, mode, params):

    x = tf.layers.dense(features, 100)
    x = tf.layers.dense(x, 100)
    predictions = tf.layers.dense(x, 100)

    rmse = tf.metrics.root_mean_squared_error(labels, predictions)

    if mode == tf.estimator.ModeKeys.PREDICT:
        predictions = {
            'predictions': predictions
        }
        return tf.estimator.EstimatorSpec(mode, predictions=predictions)

    loss = tf.losses.mean_squared_error(labels, predictions)

    metrics = {'rmse': rmse}
    print_op = tf.print((loss, rmse[1]), output_stream=sys.stdout)
    with tf.control_dependencies([print_op]):
        tf.summary.scalar('rmse', rmse[1])

    if mode == tf.estimator.ModeKeys.EVAL:
        return tf.estimator.EstimatorSpec(mode, loss=loss, eval_metric_ops=metrics)

    optimizer = tf.train.AdamOptimizer(learning_rate=1e-4)

    train_op = optimizer.minimize(loss, global_step=tf.train.get_global_step())

    return tf.estimator.EstimatorSpec(mode, loss=loss, train_op=train_op)


if __name__ == '__main__':

    import numpy as np

    features = np.random.randn(5000, 100)
    labels = features * 10 + np.random.randn(5000, 100)/1000

    x_train = {
        'features': features[:4500],
        'labels': labels[:4500]
    }

    train_input_fn = tf.estimator.inputs.numpy_input_fn(x_train['features'], x_train['labels'], batch_size=100, shuffle=True, num_epochs=300)

    x_eval = {
        'features': features[4500:],
        'labels': labels[4500:]
    }

    eval_input_fn = tf.estimator.inputs.numpy_input_fn(x_eval['features'], x_eval['labels'], batch_size=100, shuffle=False)

    config = tf.estimator.RunConfig(
        keep_checkpoint_max=1,
        model_dir='model',
        save_summary_steps=100,
        save_checkpoints_steps=100,
        log_step_count_steps=100)

    estimator = tf.estimator.Estimator(model_fn=model_fn, config=config)

    train_spec = tf.estimator.TrainSpec(input_fn=train_input_fn, max_steps=500000)
    eval_spec = tf.estimator.EvalSpec(input_fn=eval_input_fn)

    tf.estimator.train_and_evaluate(estimator, train_spec, eval_spec)
```

**Comment by oanush** [comment]
Please find the [gist](https://colab.sandbox.google.com/gist/oanush/efcfdd7112b38225f3b7f710a6db9092/32592.ipynb) of colab when tried executing the given code.Thanks!

**Comment by robertah** [comment]
I've added a random seed in the notebook. This are the loss and the rmse value plots:


<img width="376" alt="Schermata 2019-09-18 alle 10 57 19" src="https://user-images.githubusercontent.com/23496841/65133893-ba7caa00-da03-11e9-8cc5-a91b370cc7be.png">

**Comment by robertah** [comment]
So in the end I think the problem is related to `tf.losses.mean_squared_error` and `tf.metrics.mean_squared_error` that return different values during the training. Please see the notebook for more details.

**Comment by timmy-ops** [comment]
Hi @robertah,

I think I have the same problem at the moment. I get an RMSE of about 4500 instead of 900 (with a batch_size of 5). Could you help me? I dont even understand why the metrics should be divided by the batch_size...

How did you solve the problem?
